src/go/rdctl/pkg/autostart/autostart_windows.go

```diff
@@ -0,0 +1,84 @@
+package autostart
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"golang.org/x/sys/windows/registry"
+)
+
+func EnsureAutostart(autostartDesired bool) error {
+	autostartKey, err := registry.OpenKey(registry.CURRENT_USER, `Software\Microsoft\Windows\CurrentVersion\Run`, registry.SET_VALUE)
+	if err != nil {
+		return fmt.Errorf("failed to open registry key: %w", err)
+	}
+	defer autostartKey.Close()
+
+	if autostartDesired {
+		rancherDesktopPath, err := getRancherDesktopPath()
+		if err != nil {
+			return fmt.Errorf("failed to get path to Rancher Desktop: %w", err)
+		}
+		err = autostartKey.SetStringValue("RancherDesktop", rancherDesktopPath)
```

Comment:
Should the `RancherDesktop` be in a const (since we also use the string to remove it)?

Do we need to set any flags to make it not show the main window on start?

Reply:
I will put `RancherDesktop` in a `const`.

Whether the main window is shown on start is tied to `startInBackground` in the settings.